cirq/study/sweepable.py

```diff
@@ -16,17 +16,23 @@
 
 from typing import Dict, Iterable, Iterator, List, Sequence, Union, cast
 import warnings
+from typing_extensions import Protocol
 
 from cirq._doc import document
 from cirq.study.resolver import ParamResolver, ParamResolverOrSimilarType
 from cirq.study.sweeps import ListSweep, Points, Sweep, UnitSweep, Zip, dict_to_product_sweep
 
-SweepLike = Union[ParamResolverOrSimilarType, Sweep]
-document(SweepLike, """An object similar to an iterable of parameter resolvers.""")  # type: ignore
 
-Sweepable = Union[SweepLike, Iterable[SweepLike]]
+class SweepLike(Protocol):
```

Comment:
We want to make `Sweepable` recursive, not `SweepLike`. So the definition of `SweepLike` should stay as it was, and this class should be called `Sweepable`. Then we should rename the actual type to `SWEEPABLE` so the name doesn't conflict. Basically, you can copy the the `OP_TREE` template with the following substitutions:
```
Operation -> SweepLike
OP_TREE -> SWEEPABLE
```